components/common/reconcilehelper/util.go

```diff
@@ -125,7 +127,7 @@ func CopyStatefulSetFields(from, to *appsv1.StatefulSet) bool {
 		requireUpdate = true
 	}
 
-	if !reflect.DeepEqual(to.Spec.Template.Spec, from.Spec.Template.Spec) {
+	if !equality.Semantic.DeepDerivative(from.Spec.Template.Spec, to.Spec.Template.Spec) {
```

Comment:
https://github.com/kubernetes/apimachinery/blob/57f2a0733447cfd41294477d833cce6580faaca3/third_party/forked/golang/reflect/deep_equal.go#L373， pod template in statefulset we got from api-server， imagepullpolicy and some default fileds were filled， so "reflect.DeepEqual" will always be True. 
Unset fields  such as  nil pointer and empty string are ignored (not compared) in "equality.Semantic.DeepDerivative", so  "equality.Semantic.DeepDerivative" will return False when all set fileds not changed .  By the way,  it will ingore changes in these unset fields , but i think it will be better than reflect.DeepEqual.   In another way，  we can pick some fileds to compare like https://github.com/kubeflow/kubeflow/blob/7106ef6340b7f07b06a0f531f3f7a484ebb5c343/components/tensorboard-controller/controllers/tensorboard_controller.go#L480